lib/src/store.dart

```diff
@@ -199,23 +211,33 @@ class Store<State> {
   // This will be called after all other middleware provided by the user have
   // been run. Its job is simple: Run the current state through the reducer,
   // save the result, and notify any subscribers.
-  _reduceAndNotify(dynamic action) {
+  void _reduceAndNotify(dynamic action) {
     final state = reducer(_state, action);
+
+    if (distinct) {
+      if (state != _state) {
+        _updateState(state);
+      }
+    } else {
+      _updateState(state);
+    }
+  }
+
+  void _updateState(State state) {
```

Comment:
You could clean up the above by making this method:

```
void _updateState(State state) {
  if (distinct && state == _state) return;

  _state = state;
  _changeController.add(state);
}
```

Reply:
Good call.